internal/resources/metrics.go

```diff
@@ -124,13 +124,19 @@ func (m *SmbShareManager) getOrCreateMetricsService(
 	}
 	err = m.client.Create(ctx, srvWant, &rtclient.CreateOptions{})
 	if err != nil {
-		if errors.IsAlreadyExists(err) {
-			m.logger.Info("Metrics Service already exists", "key", srvKey)
-		} else {
+		if !errors.IsAlreadyExists(err) {
 			m.logger.Error(err, "Failed to create metrics Service",
 				"key", srvKey)
+			return nil, false, err
```

Comment:
IIUC the linked [issue](https://github.com/kubernetes/client-go/issues/89) discussed a different aspect where actual error returned is not understood/processed by `errors` package. It also mentions about `IsAlreadyExists` being never returned for an existing object.

For us the change here is to bypass `IsAlreadyExists` error once and hopefully retrieve next time if it got created. Is that correct?

Reply:
@anoopcs9 This PR uses the following pattern:

1) Get a resource
2) If OK, return it
3) If `!errors.IsNotFound(err)` bail-out with err
4) Create the resource
5) If `!errors.IsAlreadyExists(err)` bail-out with err
6) Get the existing resource

The linked issue hints that steps 4-5-6 could happen in cases where there is version mismatch between client-go and k8s version. Admittedly, I don't fully understand the internal's of client-go to explain *why* this could happen.

Reply:
> @anoopcs9 This PR uses the following pattern:
> 
>     1. Get a resource
> 
>     2. If OK, return it
> 
>     3. If `!errors.IsNotFound(err)` bail-out with err

Ok. Above steps were not visible in the default `git diff` view and I had to manually expand to get the complete context.